File: KIWI-Backend/src/main/java/com/d205/KIWI_Backend/order/dto/OrderRequest.java
package com.d205.KIWI_Backend.order.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class OrderRequest {
    private Integer kioskId;
    private Integer age;  // 주문자의 나이
    private Integer gender;  // 주문자의 성별
    private Integer orderNumber;

    private List<MenuOrderRequest> menuOrders;  // 여러 메뉴에 대한 주문 정보

    @Getter
    @Setter
    @Builder
    public static class MenuOrderRequest {
        private Integer menuId;  // 메뉴 ID
        private Integer quantity;  // 주문 수량
    }
}

File: KIWE-Android/data/src/main/java/com/kiwe/data/model/request/OrderRequest.kt
package com.kiwe.data.model.request

import com.kiwe.domain.model.OrderItem
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class OrderRequest(
    @SerialName("kioskId")
    val kioskId: Int = 0,
    @SerialName("age")
    val age: Int,
    @SerialName("gender")
    val gender: Int,
    @SerialName("orderNumber")
    val orderNumber: Int = 1000,
    @SerialName("menuOrders")
    val menuOrders: List<OrderItemRequest>,
)

@Serializable
data class OrderItemRequest(
    @SerialName("menuId")
    val menuId: Int,
    @SerialName("quantity")
    val quantity: Int,
)

fun OrderItem.toRequest() =
    OrderItemRequest(
        menuId = menuId,
        quantity = quantity,
    )
